src/evergreen/base.py

```diff
@@ -1,12 +1,20 @@
 # -*- encoding: utf-8 -*-
 """Task representation of evergreen."""
-from __future__ import absolute_import
+from __future__ import absolute_import, annotations
 
-from evergreen.util import parse_evergreen_datetime, parse_evergreen_date, \
-    parse_evergreen_short_datetime
+from typing import Any, Callable, Dict, Optional, TYPE_CHECKING
 
+from evergreen.util import (
+    parse_evergreen_date,
+    parse_evergreen_datetime,
+    parse_evergreen_short_datetime,
+)
 
-def evg_attrib(attrib_name, type_fn=None):
+if TYPE_CHECKING:
```

Comment:
This lets us import objects for type checking without creating circle dependencies.